core/app/models/spree/order/checkout.rb

```diff
@@ -1,324 +1,339 @@
           end
+        end
 
-          def persist_user_credit_card
-            if !self.temporary_credit_card && self.user_id && self.valid_credit_cards.present?
-              default_cc = self.valid_credit_cards.first
-              default_cc.user_id = self.user_id
-              default_cc.default = true
-              default_cc.save
-            end
+        def persist_user_credit_card
+          if !temporary_credit_card && user_id && valid_credit_cards.present?
+            default_cc = valid_credit_cards.first
+            default_cc.user_id = user_id
+            default_cc.default = true
+            default_cc.save
           end
+        end
 
-          def assign_default_credit_card
-            if self.payments.from_credit_card.count == 0 && self.user && self.user.default_credit_card.try(:valid?)
-              cc = self.user.default_credit_card
-              self.payments.create!(payment_method_id: cc.payment_method_id, source: cc)
-            end
+        def assign_default_credit_card
+          if payments.from_credit_card.count == 0 && user_has_valid_default_card? && payment_required?
+            cc = user.default_credit_card
+            payments.create!(payment_method_id: cc.payment_method_id, source: cc, amount: total)
           end
+        end
 
-          private
-          # For payment step, filter order parameters to produce the expected nested
-          # attributes for a single payment and its source, discarding attributes
-          # for payment methods other than the one selected
-          #
-          # In case a existing credit card is provided it needs to build the payment
-          # attributes from scratch so we can set the amount. example payload:
-          #
-          #   {
-          #     "order": {
-          #       "existing_card": "2"
-          #     }
-          #   }
-          #
-          def update_params_payment_source
-            if @updating_params[:payment_source].present?
-              source_params = @updating_params.delete(:payment_source)[@updating_params[:order][:payments_attributes].first[:payment_method_id].to_s]
+        def user_has_valid_default_card?
+          user && user.default_credit_card.try(:valid?)
+        end
 
-              if source_params
-                @updating_params[:order][:payments_attributes].first[:source_attributes] = source_params
-              end
+        private
+
+        # For payment step, filter order parameters to produce the expected nested
+        # attributes for a single payment and its source, discarding attributes
+        # for payment methods other than the one selected
+        #
+        # In case a existing credit card is provided it needs to build the payment
+        # attributes from scratch so we can set the amount. example payload:
+        #
+        #   {
+        #     "order": {
+        #       "existing_card": "2"
+        #     }
+        #   }
+        #
+        def update_params_payment_source
```

Comment:
Assignment Branch Condition size for update_params_payment_source is too high. [25.57/15]